### ISSUE TITLE
Frozen or saved model conversion to IR problem.

### ISSUE THREAD

**Issue by rbogdano** [openvinotoolkit/openvino#6939]
##### System information (version)
- OpenVINO => 2021.4.582
- Operating System / Platform => Ubuntu 18.04.5 LTS
- Compiler => GCC 8.4.0
- Problem classification: Model Conversion
- Framework: TensorFlow 2.5.0
- Model name: ---

##### Detailed description
I'm trying to convert frozen model or saved model to IR model.

Is there another way to do it other that writing extension to model optimizer?

python3  mo_tf.py --input_model frozen.pb  --input input_ids,input_mask,segment_ids --output bert/pooler/dense/Tanh --disable_nhwc_to_nchw --input_shape [1,128],[1,128],[1,128]
Model Optimizer arguments:
Common parameters:
        - Path to the Input Model:      /home/asd/Downloads/my_frozen/frozen.pb
        - Path for generated IR:        /home/asd/Downloads/my_frozen/.
        - IR output name:       frozen
        - Log level:    ERROR
        - Batch:        Not specified, inherited from the model
        - Input layers:         input_ids,input_mask,segment_ids
        - Output layers:        bert/pooler/dense/Tanh
        - Input shapes:         [1,128],[1,128],[1,128]
        - Mean values:  Not specified
        - Scale values:         Not specified
        - Scale factor:         Not specified
        - Precision of IR:      FP32
        - Enable fusing:        True
        - Enable grouped convolutions fusing:   True
        - Move mean values to preprocess section:       None
        - Reverse input channels:       False
TensorFlow specific parameters:
        - Input model in text protobuf format:  False
        - Path to model dump for TensorBoard:   None
        - List of shared libraries with TensorFlow custom layers implementation:        None
        - Update the configuration file with input/output node names:   None
        - Use configuration file used to generate the model with Object Detection API:  None
        - Use the config file:  None
        - Inference Engine found in:    /opt/intel/openvino_2021/python/python3.6/openvino
Inference Engine version:       2021.4.0-3839-cd81789d294-releases/2021/4
Model Optimizer version:        2021.4.0-3839-cd81789d294-releases/2021/4
2021-08-03 10:19:09.250327: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcudart.so.11.0'; dlerror: libcudart.so.11.0: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /opt/intel/openvino_2021/data_processing/dl_streamer/lib:/opt/intel/openvino_2021/data_processing/gstreamer/lib:/opt/intel/openvino_2021/opencv/lib:/opt/intel/openvino_2021/deployment_tools/ngraph/lib:/opt/intel/openvino_2021/deployment_tools/inference_engine/external/tbb/lib::/opt/intel/openvino_2021/deployment_tools/inference_engine/external/hddl/lib:/opt/intel/openvino_2021/deployment_tools/inference_engine/external/omp/lib:/opt/intel/openvino_2021/deployment_tools/inference_engine/external/gna/lib:/opt/intel/openvino_2021/deployment_tools/inference_engine/external/mkltiny_lnx/lib:/opt/intel/openvino_2021/deployment_tools/inference_engine/lib/intel64
2021-08-03 10:19:09.250349: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.
/home/asd/Documents/tensorflow/venv/lib/python3.6/site-packages/tensorflow/python/autograph/impl/api.py:22: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
  import imp
[ WARNING ]
Detected not satisfied dependencies:
        tensorflow: installed: 2.5.0, required: ~= 2.4.1

Please install required versions of components or use install_prerequisites script
/opt/intel/openvino_2021.4.582/deployment_tools/model_optimizer/install_prerequisites/install_prerequisites_tf2.sh
Note that install_prerequisites scripts may install additional components.
[ WARNING ]  Please set `version` attribute for node bert/embeddings/dropout/random_uniform/RandomUniform with type=<UNKNOWN>
[ WARNING ]  Please set `version` attribute for node bert/encoder/layer_0/attention/self/dropout/random_uniform/RandomUniform with type=<UNKNOWN>
[ WARNING ]  Please set `version` attribute for node bert/encoder/layer_0/attention/output/dropout/random_uniform/RandomUniform with type=<UNKNOWN>
[ WARNING ]  Please set `version` attribute for node bert/encoder/layer_0/output/dropout/random_uniform/RandomUniform with type=<UNKNOWN>
[ WARNING ]  Please set `version` attribute for node bert/encoder/layer_1/attention/self/dropout/random_uniform/RandomUniform with type=<UNKNOWN>
[ WARNING ]  Please set `version` attribute for node bert/encoder/layer_1/attention/output/dropout/random_uniform/RandomUniform with type=<UNKNOWN>
[ WARNING ]  Please set `version` attribute for node bert/encoder/layer_1/output/dropout/random_uniform/RandomUniform with type=<UNKNOWN>
[ WARNING ]  Changing Const node 'bert/embeddings/Slice/where_max_ends_is_needed_input_port_0/value' data type from int64 to <class 'numpy.int32'> for Elementwise operation
[ WARNING ]  Changing Const node 'bert/embeddings/dropout/random_uniform/sub' data type from float32 to <class 'numpy.int32'> for Elementwise operation
[ WARNING ]  Changing Const node 'bert/embeddings/dropout/random_uniform/min' data type from float32 to <class 'numpy.int32'> for Elementwise operation
[ WARNING ]  Changing Const node 'bert/embeddings/dropout/rate' data type from float32 to <class 'numpy.int32'> for Elementwise operation
[ WARNING ]  Changing Const node 'bert/encoder/layer_0/attention/self/dropout/random_uniform/sub' data type from float32 to <class 'numpy.int32'> for Elementwise operation
[ WARNING ]  Changing Const node 'bert/encoder/layer_0/attention/self/dropout/random_uniform/min' data type from float32 to <class 'numpy.int32'> for Elementwise operation
[ WARNING ]  Changing Const node 'bert/encoder/layer_0/attention/self/dropout/rate' data type from float32 to <class 'numpy.int32'> for Elementwise operation
[ WARNING ]  Changing Const node 'bert/encoder/layer_0/attention/output/dropout/random_uniform/sub' data type from float32 to <class 'numpy.int32'> for Elementwise operation
[ WARNING ]  Changing Const node 'bert/encoder/layer_0/attention/output/dropout/random_uniform/min' data type from float32 to <class 'numpy.int32'> for Elementwise operation
[ WARNING ]  Changing Const node 'bert/encoder/layer_0/attention/output/dropout/rate' data type from float32 to <class 'numpy.int32'> for Elementwise operation
[ WARNING ]  Changing Const node 'bert/encoder/layer_0/output/dropout/random_uniform/sub' data type from float32 to <class 'numpy.int32'> for Elementwise operation
[ WARNING ]  Changing Const node 'bert/encoder/layer_0/output/dropout/random_uniform/min' data type from float32 to <class 'numpy.int32'> for Elementwise operation
[ WARNING ]  Changing Const node 'bert/encoder/layer_0/output/dropout/rate' data type from float32 to <class 'numpy.int32'> for Elementwise operation
[ WARNING ]  Changing Const node 'bert/encoder/layer_1/attention/self/dropout/random_uniform/sub' data type from float32 to <class 'numpy.int32'> for Elementwise operation
[ WARNING ]  Changing Const node 'bert/encoder/layer_1/attention/self/dropout/random_uniform/min' data type from float32 to <class 'numpy.int32'> for Elementwise operation
[ WARNING ]  Changing Const node 'bert/encoder/layer_1/attention/self/dropout/rate' data type from float32 to <class 'numpy.int32'> for Elementwise operation
[ WARNING ]  Changing Const node 'bert/encoder/layer_1/attention/output/dropout/random_uniform/sub' data type from float32 to <class 'numpy.int32'> for Elementwise operation
[ WARNING ]  Changing Const node 'bert/encoder/layer_1/attention/output/dropout/random_uniform/min' data type from float32 to <class 'numpy.int32'> for Elementwise operation
[ WARNING ]  Changing Const node 'bert/encoder/layer_1/attention/output/dropout/rate' data type from float32 to <class 'numpy.int32'> for Elementwise operation
[ WARNING ]  Changing Const node 'bert/encoder/layer_1/output/dropout/random_uniform/sub' data type from float32 to <class 'numpy.int32'> for Elementwise operation
[ WARNING ]  Changing Const node 'bert/encoder/layer_1/output/dropout/random_uniform/min' data type from float32 to <class 'numpy.int32'> for Elementwise operation
[ WARNING ]  Changing Const node 'bert/encoder/layer_1/output/dropout/rate' data type from float32 to <class 'numpy.int32'> for Elementwise operation
[ ERROR ]  List of operations that cannot be converted to Inference Engine IR:
[ ERROR ]      RandomUniform (7)
[ ERROR ]          bert/embeddings/dropout/random_uniform/RandomUniform
[ ERROR ]          bert/encoder/layer_0/attention/self/dropout/random_uniform/RandomUniform
[ ERROR ]          bert/encoder/layer_0/attention/output/dropout/random_uniform/RandomUniform
[ ERROR ]          bert/encoder/layer_0/output/dropout/random_uniform/RandomUniform
[ ERROR ]          bert/encoder/layer_1/attention/self/dropout/random_uniform/RandomUniform
[ ERROR ]          bert/encoder/layer_1/attention/output/dropout/random_uniform/RandomUniform
[ ERROR ]          bert/encoder/layer_1/output/dropout/random_uniform/RandomUniform
[ ERROR ]  Part of the nodes was not converted to IR. Stopped.
 For more information please refer to Model Optimizer FAQ, question #24. (https://docs.openvinotoolkit.org/latest/openvino_docs_MO_DG_prepare_model_Model_Optimizer_FAQ.html?question=24#question-24)

##### Steps to reproduce
1. Downloading https://tfhub.dev/google/small_bert/bert_uncased_L-2_H-256_A-4/2?tf-hub-format=compressed
2. Converting saved model to frozen using below command
freeze_graph.freeze_graph(saved_model_tags = tag_constants.TRAINING, input_saved_model_dir="small_bert_uncased", input_graph=None, input_saver=False, input_binary=False, input_checkpoint=None, output_node_names='bert/pooler/dense/Tanh', restore_op_name=None, filename_tensor_name=None, output_graph="frozen.pb", clear_devices=True, initializer_nodes='')
3. Trying to convert to IR using below command
python3 mo_tf.py --input_model frozen.pb  --input input_ids,input_mask,segment_ids --output bert/pooler/dense/Tanh --disable_nhwc_to_nchw --input_shape [1,128],[1,128],[1,128]

As I mention I also tried to do this without converting it to frozen model with the same result

**Comment by zulkifli-halim** [comment]
Hi @rbogdano, It seems like some arguments are missing when converting the model, or there is something wrong with the model itself.  

We do have the following Intel pre-trained models that might suit your needs ;

[bert-large-uncased-whole-word-masking-squad-0001](https://community.intel.com/t5/Intel-Distribution-of-OpenVINO/Convert-BERT-TF-model-pb-to-IR/td-p/1301717)
[bert-large-uncased-whole-word-masking-squad-emb-0001](https://docs.openvinotoolkit.org/2021.4/omz_models_model_bert_large_uncased_whole_word_masking_squad_emb_0001.html)
[bert-large-uncased-whole-word-masking-squad-int8-0001](https://docs.openvinotoolkit.org/2021.4/omz_models_model_bert_large_uncased_whole_word_masking_squad_int8_0001.html)
[bert-small-uncased-whole-word-masking-squad-0001](https://docs.openvinotoolkit.org/2021.4/omz_models_model_bert_small_uncased_whole_word_masking_squad_0001.html)
[bert-small-uncased-whole-word-masking-squad-0002](https://docs.openvinotoolkit.org/2021.4/omz_models_model_bert_small_uncased_whole_word_masking_squad_0002.html)
[bert-small-uncased-whole-word-masking-squad-emb-int8-0001](https://docs.openvinotoolkit.org/2021.4/omz_models_model_bert_small_uncased_whole_word_masking_squad_emb_int8_0001.html)
[bert-small-uncased-whole-word-masking-squad-int8-0002](https://docs.openvinotoolkit.org/2021.4/omz_models_model_bert_small_uncased_whole_word_masking_squad_int8_0002.html)


And also the following public pre-trained model:

[bert-base-ner](https://github.com/openvinotoolkit/open_model_zoo/tree/master/models/public/bert-base-ner)

**Comment by rbogdano** [comment]
Ok I guess your right. I workaround this problem by using different model. Thanks!
